examples_tests/20.Megatexture/main.cpp

```diff
@@ -17,18 +17,34 @@ using namespace core;
 constexpr const char* GLSL_VT_DESCRIPTORS = //also turns off set3 bindings (textures) because they're not needed anymore as we're using VT
 R"(
 #ifndef _NO_UV
+
+#define _IRR_VT_DESCRIPTOR_SET 0
+#define _IRR_VT_PAGE_TABLE_BINDING 0
+
+#define _IRR_VT_FLOAT_VIEWS_BINDING 1 
+#define _IRR_VT_FLOAT_VIEWS_COUNT 3
+#define _IRR_VT_FLOAT_VIEWS
+
+#define _IRR_VT_INT_VIEWS_BINDING 2
+#define _IRR_VT_INT_VIEWS_COUNT 1
+#define _IRR_VT_INT_VIEWS
+
+#define _IRR_VT_UINT_VIEWS_BINDING 3
+#define _IRR_VT_UINT_VIEWS_COUNT 0
+#define _IRR_VT_UINT_VIEWS
+#include <irr/builtin/glsl/virtual_texturing/descriptors.glsl>
+
 #define VT_COUNT 3
-#include <%s> //irr/builtin/glsl/virtual_texturing/descriptors.glsl/...
 layout (set = 2, binding = 1, std430) readonly restrict buffer LUT
 {
-    uint lut[VT_COUNT];
+    uint lut[];
 } layer2pid;
 
-layout (set = 2, binding = 0, std430) readonly restrict buffer PrecomputedStuffUBO
+layout (set = 2, binding = 0, std140) uniform PrecomputedStuffUBO
 {
-    uint pgtab_sz_log2[VT_COUNT];
+    uint pgtab_sz_log2;
+    float vtex_sz_rcp;
     float phys_pg_tex_sz_rcp[VT_COUNT];
```

Comment:
put it together in one SSBO, you'll have less alignment issues (`phys_pg_tex_sz_rcp` has a stride of 16 bytes under UBO packing)